[🐛 Bug]: Typing issues with Jasmine integration

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.20.5

### Node.js Version

18.18.2

### Mode

WDIO Testrunner

### Which capabilities are you using?

_No response_

### What happened?

In [our project](https://github.com/DataDog/browser-sdk/) we are using the WDIO testrunner with TypeScript and Jasmine integration. To use the correct TS typings, we include types in a [specific order](https://github.com/DataDog/browser-sdk/blob/94be1ed99a5a383adb1a01d7086d995e95455c6d/test/e2e/tsconfig.json#L11), so the types from `@wdio/global/types` are partially overriden by the `jasmine` types, especially for the `expect` global variable.

When upgrading from WDIO 8.20.0 to 8.20.5, our test suite fails with typing errors because the `expect` global variable has the Jest `expect` type, not the Jasmine one. I believe this is because of PR https://github.com/webdriverio/webdriverio/pull/11524, which makes the wdio global types take precedence over jasmine types.


### What is your expected behavior?

I would expect to be able to use the correct jasmine types.

Would it be possible to revert the way it was before 8.20.5? Else, would it be possible to have documentation on how to setup the Jasmine integration with TypeScript (the [current documentation](https://webdriver.io/docs/frameworks#using-jasmine) does not mention types)?

### How to reproduce the bug.

* Clone https://github.com/webdriverio/jasmine-boilerplate/
* Add `expect(true).toBeTrue()` in one of the spec ([jasmine doc](https://jasmine.github.io/api/5.1/matchers.html#toBeTrue))
* Run `npx tsc . --noEmit true` to typecheck the project


### Relevant log output

```typescript
test/specs/form.spec.ts:26:22 - error TS2339: Property 'toBeTrue' does not exist on type 'Matchers<void | Promise<void>, boolean>'.

26         expect(true).toBeTrue();
                        ~~~~~~~~


Found 1 error in test/specs/form.spec.ts:26
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

--- a/packages/wdio-jasmine-framework/src/index.ts
+++ b/packages/wdio-jasmine-framework/src/index.ts
@@ -512,4 +512,8 @@ declare global {
     namespace WebdriverIO {
         interface JasmineOpts extends jasmineNodeOpts {}
     }
+    namespace ExpectWebdriverIO {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        interface Matchers<R, T> extends jasmine.Matchers<R> {}
+    }
 }

--- a/tests/typings/jasmine/config.ts
+++ b/tests/typings/jasmine/config.ts
@@ -12,3 +12,7 @@ const config: WebdriverIO.Config = {
  * check import of assertion lib
  */
 expect($('foo')).toHaveTextContaining('foobar')
+/**
+ * check support for Jasmine specific matchers
+ */
+expect(true).toBeTrue()

